Polish settings UI localized text

carlos-zamora opened the pull request:
Updates the following text in the settings UI
- focus follow mouse mode is introduced to be more instructional
- focus follow mouse mode tooltip removed
- avoid double negative in "disable pane animation"

## Demo

Focus Follow Mouse Mode was renamed to be more similar to CopyOnSelect's design ("Automatically copy selection to clipboard" with no tooltip).
![focus follow mouse mode](https://user-images.githubusercontent.com/11050425/107697117-ab854000-6c67-11eb-9f60-2cafb0473782.png)

"Disable pane animations":
![Pane animations](https://user-images.githubusercontent.com/11050425/107697446-13d42180-6c68-11eb-9b26-33614b84422b.png)



Closes #8900 
Updates #6459 Settings UI text

carlos-zamora commented:
WAIT. I need to actually make this converter go both ways!

DHowett commented on src/cascadia/TerminalSettingsEditor/Converters.idl:
```diff
@@ -14,6 +14,11 @@ namespace Microsoft.Terminal.Settings.Editor
         FontWeightConverter();
     };
 
+    runtimeclass InvertedBooleanConverter : [default] Windows.UI.Xaml.Data.IValueConverter
```
how many of these freaking converters do we need lol

carlos-zamora replied:
_yeaaaah_. This is just...dumb... We could make this all a part of the view model. But like, this is so much more straightforward and I have a feeling we'll need it again at some point haha

DHowett replied:
We have introduced another whole converter _and refactored the converter code_ just for this highly controversial string.

DHowett commented on src/cascadia/TerminalSettingsEditor/Resources/en-US/Resources.resw:
```diff
@@ -641,7 +641,7 @@
     <value>None</value>
   </data>
   <data name="Globals_DisableAnimations.Header" xml:space="preserve">
```
you can't do this. if I cut a servicing build of 1.6 after you make this change, suddenly the text will change but the logic will not.

carlos-zamora replied:
So, that means that this just can't be a fix that's serviced, right? It'll have to be a part of v1.7?

DHowett replied:
I just put out a servicing update, I'm not going to flip around folks' booleans just days after making them update to 1.6.1041x ;P
What I mean is that changing this for 1.7 will break the key historically for 1.6 builds, BECAUSE it is the same key.

zadjii-msft replied:
So, should we block this PR on renaming this key?

carlos-zamora replied:
I think this should be a part of v1.7 because...
- Settings UI is in preview, so this is the time we _can_ make that change (smaller audience)
- Settings UI is also hidden behind a keybinding users have to opt-in for, so we have an even less reduced number of affected users. Once we include it in defaults.json, we don't have that luxury
- we got explicit feedback from a user saying that this was an issue

We should make sure this change is excluded from any servicing builds prior to v1.7 though. But that's a normal thing to do.

DHowett commented on src/cascadia/TerminalSettingsEditor/GlobalAppearance.xaml:
```diff
@@ -61,7 +63,7 @@ the MIT License. See LICENSE in the project root for license information. -->
             <!--Disable Animations-->
             <local:SettingContainer x:Uid="Globals_DisableAnimations">
-                <ToggleSwitch IsOn="{x:Bind State.Globals.DisableAnimations, Mode=TwoWay}"/>
+                <ToggleSwitch IsOn="{x:Bind State.Globals.DisableAnimations, Mode=TwoWay, Converter={StaticResource InvertedBooleanConverter}}"/>
```
I disagree fundamentally with this change. I'm not blocking you over it, but I don't think human interface guidebooks are bibles that we should sleep with under our pillows.

carlos-zamora replied:
_Technically_, the problem is that we should've named this setting something else. But that's much harder to fix (or we can do some kind of enum design, but I think _nobody_ wants that hahaha).

I do get what you're saying though. I think we just need to be more careful with these kinds of settings in the future. :/

DHowett replied:
I really do not think we need to be more careful. It's the settings UI's job to properly represent whatever the underlying data model stores in whatever way is best for the user. "Disabling" things makes sense in the settings.

It's going to be far more annoying when we talk about "disabled" profile sources.

carlos-zamora replied:
Here's a setting I found in an app on my phone the other day:
![example](https://user-images.githubusercontent.com/11050425/108110244-d85e9c00-7047-11eb-80cd-2a2d4f96f1fd.png)
I didn't go out looking for this haha. I just ran into it and am now eternally confused by what that switch does. Btw, the gray text does _not_ update when you switch the value.

It's a real issue. We do need to be more careful.

Disabling profile sources is an issue for a separate discussion.

DHowett replied:
I consider this switch to be on par with iOS's "Reduce motion" switch. It's on: motion is reduced. It is a switch that disables a thing when it is on.

cinnamon-msft replied:
Chiming in here, I don't believe this is on par with "Reduce motion", since the switch is "enabling the reduction of motion" rather than "enabling the disabling of motion".

I think it makes sense to make the change to "Pane animations" and to do it now rather than wait for this to become the default settings experience.

DHowett replied:
I mean, if we changed the wording to "reduce motion" it would 1. mean exactly the same thing and 2. do exactly the same thing (reduce motion), so... maybe we should do that? :smile:

cinnamon-msft replied:
My concern with "reduce motion" is that there's an arbitrary toggle switch on the Appearance page that just says "reduce motion" without much context.

I'm trying to think of other wording that describes what happens when pane animations are disabled, that would help prevent switching the toggle logic. Something like "snap open panes"..?

zadjii-msft commented on src/cascadia/TerminalSettingsEditor/InvertedBooleanConverter.h:
```diff
@@ -0,0 +1,30 @@
+#include "InvertedBooleanConverter.g.h"
+#include "Utils.h"
+
+namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
+{
+    struct InvertedBooleanConverter : InvertedBooleanConverterT<InvertedBooleanConverter>
+    {
+        InvertedBooleanConverter() = default;
+
+        Windows::Foundation::IInspectable Convert(Windows::Foundation::IInspectable const& value,
+                                                  Windows::UI::Xaml::Interop::TypeName const& targetType,
+                                                  Windows::Foundation::IInspectable const& parameter,
+                                                  hstring const& language);
+
+        Windows::Foundation::IInspectable ConvertBack(Windows::Foundation::IInspectable const& value,
+                                                      Windows::UI::Xaml::Interop::TypeName const& targetType,
+                                                      Windows::Foundation::IInspectable const& parameter,
+                                                      hstring const& language);
+    };
+}
+
+namespace winrt::Microsoft::Terminal::Settings::Editor::factory_implementation
+{
+    BASIC_FACTORY(InvertedBooleanConverter);
+}
```
```suggestion
#include "../inc/cppwinrt_utils.h"

DECLARE_CONVERTER(winrt::Microsoft::Terminal::Settings::Editor, InvertedBooleanConverter);
```

carlos-zamora replied:
Changed. Should I file a follow-up task to throw all of the converters in a single `Converters.h` and `Converters.cpp` pair? Just a random thought. Noticed "StringIsNotDesktopConverter" has two converters in it.

zadjii-msft commented on src/cascadia/TerminalSettingsEditor/InvertedBooleanConverter.h:
```diff
@@ -0,0 +1,30 @@
+#pragma once
+
+#include "InvertedBooleanConverter.g.h"
+#include "Utils.h"
```
Utils.h? Why Utils.h?

zadjii-msft reviewed (approved):
You know I'm fine with this PR, but it sounds like Dustin actually wants a block? I'm not sure I totally follow that discussion. 
I concur that I don't _really_ agree with flipping the wording around, but I also don't care very strongly about it.

DHowett reviewed (changes requested):
Noted above in discussion. Regardless of whether we're flipping the sense of a single switch because our users cannot handle true falses, _you cannot recycle the key and change its meaning_ because when I make a build of 1.6 it pulls the latest strings from the translation pack and oops-if-it-happens-to-be-updated-with-the-opposite-text-i-guess-we-done-messed-up.

DHowett commented:
If you want this in 1.7 it has to go in TODAY.

carlos-zamora commented:
@DHowett I kept the old key untouched for when older WT versions need it. And I introduced "Globals_DisablePaneAnimationsReversed" and added a comment to track it.

> Why not "Globals_PaneAnimations"?

Good question random person that didn't ask that question! I decided to keep the "_DisablePaneAnimations"  part because I want the structure of these UIDs to be the same: "<Context>_<Setting><AdditionalInfo>".
